### PR TITLE
feat: React PFD Rework

### PR DESCRIPTION
**Pull request by lukecologne**
## Summary of Changes

* A completely new PFD created in React.js, for better visual fidelity, and hopefully better performance.

Credit to https://bitbarrelmedia.wordpress.com/2017/11/01/real-time-efis-vector-graphics/ for reference images.

## Additional context

Discord username (if different from GitHub):

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

File: src/instruments/src/PFD/AttitudeIndicatorHorizon.jsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+const RisingGround = ({ radioAlt, pitch }) => {
+    const targetPitch = -0.1 * radioAlt;
+
+    const targetOffset = Math.max(Math.min(calculateHorizonOffsetFromPitch(pitch - targetPitch) - 31.563, 0), -63.093);
+
+    return [
+        <path d="m113.95 157.74h-90.08v-45.357h90.08z" transform={`translate(0 ${targetOffset})`} className="NormalOutline EarthFill" />,
+        <path id="HorizonGroundRectangle" d="m113.95 157.74h-90.08v-45.357h90.08z" transform={`translate(0 ${targetOffset})`} className="NormalStroke White" />,
+    ];
```
Comment (beheh) [inline src/instruments/src/PFD/AttitudeIndicatorHorizon.jsx:191]: I'd suggest using React fragments if possible over arrays, and only use arrays when you're truly working with a dynamic list where the length can change:

```suggestion
    return <>
        <path d="m113.95 157.74h-90.08v-45.357h90.08z" transform={`translate(0 ${targetOffset})`} className="NormalOutline EarthFill" />,
        <path id="HorizonGroundRectangle" d="m113.95 157.74h-90.08v-45.357h90.08z" transform={`translate(0 ${targetOffset})`} className="NormalStroke White" />,
    </>;
```

File: src/instruments/src/PFD/HeadingIndicator.jsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        <text id="SelectedHeadingTextLeft" className="FontSmallest MiddleAlign Cyan" x="36.20676" y="144.44794">{text}</text>
+    );
+};
+
+const GroundTrackBug = ({ heading, groundTrack }) => {
+    const offset = getSmallestAngle(groundTrack, heading) * DistanceSpacing / ValueSpacing;
+    return [
+        <path className="ThickOutline" transform={`translate(${offset} 0)`} d="m68.906 145.75-1.2592 1.7639 1.2592 1.7639 1.2592-1.7639z" />,
+        <path id="ActualTrackIndicator" className="ThickStroke Green" transform={`translate(${offset} 0)`} d="m68.906 145.75-1.2592 1.7639 1.2592 1.7639 1.2592-1.7639z" />,
+    ];
```
Comment (beheh) [inline src/instruments/src/PFD/HeadingIndicator.jsx:105]: Same here, I recommend fragments:

```suggestion
    return <>
        <path className="ThickOutline" transform={`translate(${offset} 0)`} d="m68.906 145.75-1.2592 1.7639 1.2592 1.7639 1.2592-1.7639z" />,
        <path id="ActualTrackIndicator" className="ThickStroke Green" transform={`translate(${offset} 0)`} d="m68.906 145.75-1.2592 1.7639 1.2592 1.7639 1.2592-1.7639z" />,
    </>;
```

File: src/instruments/src/PFD/AttitudeIndicatorHorizon.jsx (hunk 1 above)
Comment (lukecologne) [reply src/instruments/src/PFD/AttitudeIndicatorHorizon.jsx:191]: Ah, didn't know this was possible, I'll change this as soon as I can. Thanks!

File: src/instruments/src/PFD/HeadingIndicator.jsx (hunk 2 above)
Comment (lukecologne) [reply src/instruments/src/PFD/HeadingIndicator.jsx]: Alright, I think I got all of them

File: src/instruments/src/PFD/index.jsx [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        const roll = getSimVar('PLANE BANK DEGREES', 'degrees');
+        const heading = getSimVar('PLANE HEADING DEGREES MAGNETIC', 'degrees');
+
+        if (!this.isAttExcessive && (pitch > 25 || pitch < -13 || Math.abs(roll) > 45)) {
+            this.isAttExcessive = true;
+        } else if (this.isAttExcessive && pitch < 22 && pitch > -10 && Math.abs(roll) < 40) {
+            this.isAttExcessive = false;
+        }
+
+        const groundTrack = getSimVar('GPS GROUND MAGNETIC TRACK', 'degrees');
```
Comment (beheh) [inline src/instruments/src/PFD/index.jsx:95]: As this seems to report weird values near standstill I would suggest maybe just overriding this with the magnetic heading below a certain airspeed. E.g. compare https://www.youtube.com/watch?v=n3TJMro_NzA#t=1m28s (especially the pushback with a "backwards" ground track) with https://discord.com/channels/738864299392630914/739223625898655854/809194933247672360. In the reference footage, during takeoff, the two only seem to diverge starting at 50-60 knots.

**Comment by lukecologne** [comment]
Closing this since it is integrated in the autopilot branch now.

**Pull request closed without merging**